sdk/tables/data-tables/src/TablesBatch.ts

```diff
@@ -0,0 +1,321 @@
+      }
+      interceptClient.updateEntity(entity, mode, options);
+    },
+    async submitBatch(): Promise<any> {
+      const body = batchRequest.getHttpRequestBody();
+      const client = new ServiceClient();
+      const headers: RawHttpHeaders = {
+        accept: "application/json",
+        "x-ms-version": "2019-02-02",
+        "Accept-Charset": "UTF-8",
+        DataServiceVersion: "3.0;",
+        MaxDataServiceVersion: "3.0;NetFx",
+        "Content-Type": `multipart/mixed; boundary=batch_${batchGuid}`,
+        Connection: "Keep-Alive"
+      };
+
+      const request = new WebResource(batchUrl, "POST", body, undefined, new HttpHeaders(headers));
+
+      if (credential) {
+        const authHeader = getAuthorizationHeader(request, credential);
+        request.headers.set("Authorization", authHeader);
+      }
+
+      const requestOptions: RequestPrepareOptions = {
+        method: "POST",
+        url: batchUrl,
+        headers: request.headers.rawHeaders(),
+        body,
+        disableJsonStringifyOnBody: true
+      };
+      console.log(body);
+      const rawBatchResponse = await client.sendRequest(requestOptions);
+      console.log(rawBatchResponse.bodyAsText);
+      return rawBatchResponse;
+    }
+  };
+}
+
+/**
+ * This method creates a batch request object that provides functions to build the envelope and body for a batch request
+ * @param batchGuid Id of the batch
+ */
+function createInnerBatchRequest(batchGuid: string) {
+  const HTTP_LINE_ENDING = "\r\n";
+  const HTTP_VERSION_1_1 = "HTTP/1.1";
+  const changesetGuid = generateUuid();
+
+  // batch_{batchid}
+  const batchBoundary = `batch_${batchGuid}`;
+  const changesetBoundary = `changeset_${changesetGuid}`;
+
+  const subRequestPrefix = `--${changesetBoundary}${HTTP_LINE_ENDING}${HeaderConstants.CONTENT_TYPE}: application/http${HTTP_LINE_ENDING}${HeaderConstants.CONTENT_TRANSFER_ENCODING}: binary`;
+  // Content-Type: multipart/mixed; boundary=changeset_{changesetGuid}
+  const multipartContentType = `multipart/mixed; boundary=changeset_${changesetGuid}`;
+  const chaangesetEnding = `--${changesetBoundary}--`;
+  const batchEnding = `--${batchBoundary}`;
+
+  return {
+    body: `--${batchBoundary}${HTTP_LINE_ENDING}${HeaderConstants.CONTENT_TYPE}: multipart/mixed; boundary=changeset_${changesetGuid}${HTTP_LINE_ENDING}${HTTP_LINE_ENDING}`,
+    operationCount: 0,
+    createPipeline() {
+      const policyFactoryLength = 3;
+      const factories: RequestPolicyFactory[] = new Array(policyFactoryLength);
+
+      factories[0] = deserializationPolicy();
+      factories[1] = new BatchHeaderFilterPolicyFactory();
+
+      factories[policyFactoryLength - 1] = new BatchRequestAssemblePolicyFactory(this); // Use batch assemble policy to assemble request and intercept request from going to wire
+      return factories;
+    },
+    appendSubRequestToBody(request: WebResource) {
+      // Start to assemble sub request
+      this.body += [
```

Comment:
I'd make the subrequest string and **then** append it to body, for readability. It might also be worthwhile to have the body be an array of strings until a final join before sending.